fastai/losses.py

```diff
@@ -122,4 +122,24 @@ class LabelSmoothingCrossEntropyFlat(BaseLoss):
     @use_kwargs_dict(keep=True, eps=0.1, reduction='mean')
     def __init__(self, *args, axis=-1, **kwargs): super().__init__(LabelSmoothingCrossEntropy, *args, axis=axis, **kwargs)
     def activation(self, out): return F.softmax(out, dim=-1)
-    def decodes(self, out):    return out.argmax(dim=-1)
\ No newline at end of file
+    def decodes(self, out):    return out.argmax(dim=-1)
+
+# Cell
+def _one_hot(x, classes, axis=1):
+    "Target mask to one hot"
+    return torch.stack([torch.where(x==c, 1,0) for c in range(classes)], axis=axis)
+
+# Cell
+class DiceLoss:
+    "Dice loss for segmentation"
+    def __init__(self, axis=1, smooth=1):
+        store_attr()
+    def __call__(self, pred, targ):
+        targ = _one_hot(targ, pred.shape[self.axis])
```

Comment:
Why not use the `one_hot` from `torch_core`?

Reply:
I know we have one, but is does not work the same. I would need to modify the other one to be able to create the one-hot segmentation masks.

Reply:
the `DiceMulti` metric already does this:
```python
        for c in range(learn.pred.shape[self.axis]):
            p = torch.where(pred == c, 1, 0)
```

Reply:
What's the difference needed in functionality exactly? Is there some different name you could use for yours that makes it clear? And/or some comment you could add?

Reply:
I could make it a class method, I just put it outside to test it =P